mysite/views.py

```diff
@@ -1,3 +1,175 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect, get_object_or_404
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login, get_user_model
+from django.http import HttpResponseRedirect
+from django import forms
+from django.urls import reverse
+from django.views.generic import UpdateView
+from django.contrib import messages
+from django.contrib.auth import update_session_auth_hash
+from django.contrib.auth.forms import PasswordChangeForm
+from mysite.forms import UserRegistrationForm, UserProfileForm, ProjectForm, ProjectBidForm, AssignProjectForm
+from mysite.models import Project, ProjectBid
 
-# Create your views here.
+User = get_user_model()
+
+
+def home(request):
+    return render(request, 'mysite/home.html')
+
+
+def register(request):
+    if request.method == 'POST':
+        form = UserRegistrationForm(request.POST)
+        if form.is_valid():
+            user_obj = form.cleaned_data
+            username = user_obj['username']
+            email = user_obj['email']
+            password = user_obj['password']
+            user_role = user_obj['user_role']
+            if not (User.objects.filter(username=username).exists() or User.objects.filter(email=email).exists()):
+                User.objects.create_user(username, email, password)
+                user = authenticate(username = username, password = password)
+                # Need recommendation
+                user.user_role = user_role
+                user.save()
+                login(request, user)
+                return HttpResponseRedirect('/')
+            else:
+                raise forms.ValidationError('Looks like a username with that email or password already exists')
```

Comment:
Message is giving incorrect information as we are searching use against username and email. It should be update to 
'Looks like a user with that email or username already exists'

Reply:
Feedback Incorporated.